src/wp-includes/option.php

```diff
@@ -956,6 +956,24 @@ function get_transient( $transient ) {
  */
 function set_transient( $transient, $value, $expiration = 0 ) {
```

Comment:
You add the limitations only in `set_transient()`, which I agree is the root of the problem. But additionally, wouldn't it make sense to enforce those limitations in the other transient functions too? Potentially we can introduce a new private function like `wp_validate_transient_name()` and reuse in the relevant places?

Reply:
+1

Reply:
+1, I'd also suggest that we don't perform a check for an empty value until after we add a call to `trim( (string) $transient ) )` on the name, at which point we can check `'' === $transient` and bail, with an error preferably.

Reply:
Adding my 2 cents:

I think this fix only needs to deal with checking for empty string. The function is clearly documented that the transient name must be a string. Anything other data type, such as `integer`, `boolean,` or `null`, are doing it wrong.

If all it needs is to check for `is_string()` and `empty()` string, then thinking a separate validation function might be overkill.